route.go

```diff
@@ -147,53 +188,108 @@ func (route *baseRoute) Shutdown() error {
 }
 
 // to view the state of the table/route at any point in time
-func (route *RouteSendAllMatch) Snapshot() RouteSnapshot {
+func makeSnapshot(route *baseRoute, routeType string) RouteSnapshot {
 	conf := route.config.Load().(RouteConfig)
-	dests := make([]*Destination, len(conf.Dests))
-	for i, d := range conf.Dests {
+	dests := make([]*Destination, len(conf.Dests()))
+	for i, d := range conf.Dests() {
 		dests[i] = d.Snapshot()
 	}
-	return RouteSnapshot{conf.Matcher, dests, "sendAllMatch", route.key}
+	return RouteSnapshot{*conf.Matcher(), dests, routeType, route.key}
+
+}
+
+func (route *RouteSendAllMatch) Snapshot() RouteSnapshot {
+	return makeSnapshot(&route.baseRoute, "sendAllMatch")
 }
 
 func (route *RouteSendFirstMatch) Snapshot() RouteSnapshot {
-	conf := route.config.Load().(RouteConfig)
-	dests := make([]*Destination, len(conf.Dests))
-	for i, d := range conf.Dests {
-		dests[i] = d.Snapshot()
-	}
-	return RouteSnapshot{conf.Matcher, dests, "sendFirstMatch", route.key}
+	return makeSnapshot(&route.baseRoute, "sendFirstMatch")
+}
+
+func (route *RouteConsistentHashing) Snapshot() RouteSnapshot {
+	return makeSnapshot(&route.baseRoute, "consistentHashing")
+}
+
+// baseConfigExtender is a function that takes a baseRouteConfig and returns
+// a configuration object that implements RouteConfig. This function may be
+// the identity function, i.e., it may simply return its argument.
+// This mechanism supports maintaining different configuration objects for
+// different route types and creating a route-appropriate configuration object
+// on a configuration change.
+// The baseRoute object implements private methods (addDestination,
+// delDestination, etc.) that create a new baseRouteConfig object that reflects
+// the configuration change. This baseRouteConfig object is applicable to all
+// route types. However, some route types, like ConsistentHashingRoute, have
+// additional configuration and therefore have a distinct configuration object
+// that embeds baseRouteConfig (in the case of ConsistentHashingRoute, the object
+// is consistentHashingRouteConfig). This route-type-specific configuration also
+// needs to be updated on a base configuration change (e.g., on a change
+// affecting destinations). Accordingly, the public entry points that effect the
```

Comment:
I really did mean effect as a verb in this case. :) http://dictionary.reference.com/browse/effect, meaning 10.